cloudigrade/account/reports/__init__.py

```diff
@@ -179,3 +187,113 @@ def _calculate_instance_usage(start, end, events):
         time_running += diff.total_seconds()
 
     return time_running
+
+def validate_event(event, start):
+    """
+    Ensure that the event is relevant to our time frame.
+
+    Args:
+        event: (InstanceEvent): The event object to evaluate
+        start (datetime.datetime): Start time (inclusive)
+
+    Returns:
+        bool: A boolean regarding whether or not we should inspect the event further.
+    """
+    valid_event = True
+    # if the event occurred outside of our specified period (ie. before start) we should
+    # only inspect it if it was a power on event
+    if event.occurred_at < start:
+        if event.event_type == InstanceEvent.TYPE.power_off:
+            valid_event = False
+    return valid_event
+
+def validate_account_creation(account, end):
+    """
+    Ensure that the event is relevant to our time frame.
+
+    Args:
+        account: (Account): The event object to evaluate
+        end (datetime.datetime): End time (exclusive)
+
+    Returns:
+        bool: A boolean regarding whether or not the account was created after the end
+        time.
+    """
+    if end <= account.created_at:
+        return False
+    return True
+
+
+def get_account_overview(account, start, end):
+    """
+    Generate an overview of an account over a specified amount of time.
+
+    Args:
+        start (datetime.datetime): Start time (inclusive)
+        end (datetime.datetime): End time (exclusive)
+        account (Account or AwsAccount): Account or AwsAccount object
+
+    Returns:
+        dict: An overview of the instances/images/rhel & openshift images for the specified
+        account during the specified time period."""
+    machine_images = AwsMachineImage.objects.all()
+    instances = []
+    images = []
+    rhel = []
+    openshift = []
+    cloud_type = 'unknown'
+    arn = 'unsupported'
+    # make sure that this is a supported cloud type (aws at this time)
+    if isinstance(account, AwsAccount):
+        cloud_helper = helper.get_report_helper(AWS_PROVIDER_STRING, account.aws_account_id)
+        # _get_relevant_events will return the events in between the start & end times & if
+        # no events are present during this period, it will return the last event that occurred
+        events = _get_relevant_events(start, end, cloud_helper)
+        for event in events:
+            valid_event = validate_event(event, start)
+            if valid_event:
+                instances.append(event.instance.id)
+                images.append(event.ec2_ami_id)
+                image = machine_images.filter(ec2_ami_id=event.ec2_ami_id).first()
```

Comment:
...to use the model's objects directly like `AwsMachineImage.objects.filter(...).first()`.

Or even better, you can simplify that to `AwsMachineImage.objects.get(...)` to get exactly the one match. If no match is found, `get` will raise an exception, but that's okay here because we really _should_ have that image saved at this point.

(If we merge cloudigrade/pull/405, this line can go away entirely since you will be able to get the image at `event.machineimage`.)